### PR TITLE
[MOV] web: split controllers.main in several files

### PR DESCRIPTION
**Pull request by Julien00859**
The odoo.addons.web.controllers.main module was a very long bloated file where many different controllers were concatened. It proved difficult to work on that file on a regular basis,mainly because ctrl-p "web main.py" was not pointing the right file.

In this work the file has been split on the basic 1 controller = 1 file. The original way of importing stuff (through main.py) is still possible thanks to deprecated aliases.

### CODE REVIEW

**Comment by Julien00859** [comment]
@robodoo override=ci/style

I corrected most of the problems reported by ci/style, the one left are `W0622(redefined-builtin) coll 64: Redefining built-in 'id' ` where `id` is part of a public API. There is no value in breaking a public API just so that a linter is happy. The same goes for using `ICP.get_param('web.base.url')`, the linter is about not adding new code that uses that ICP, moving existing code that uses it is fine.

**Comment by ged-odoo** [comment]
looks good to me

**Comment by rco-odoo** [comment]
@robodoo override=ci/security

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

File: addons/web/controllers/main.py [hunk 1]
```diff
@@ -1,1992 +1,55 @@
...
-    'sum': sum,
-    'bool_and': all,
-    'bool_or': any,
+from odoo.tools import lazy
+from odoo.addons.web.controllers import (
+    action, binary, database, dataset, export, home, report, session,
+    utils, view, webclient,
+)
+
+_MOVED_TO_MAP = {
```
Comment (odony) [inline addons/web/controllers/main.py:11]: Thanks for maintaining the compatibility! 👍👍

I'm wondering: is the `_MOVED_TO_MAP` + module-level `getattr` really better than a plain list of assignations? It has slightly less redundancy, but overall it feels less readable, and the `getattr` magic could backfire / shadow things, no?


```
_get_login_redirect_url = utils._get_login_redirect_url
_local_web_translations = utils._local_web_translations
...
```

**Comment by odony** [comment]
> override=ci/security

@rco-odoo This normally requires a small rationale explanation for posterity. If the CI build log is gone in 6 months, how do we know which flag were raised, and why were they safe to bypass? e.g. here, was this strictly caused by moving existing code? (hint: no it's not)
And even if the code was strictly moved around, is it still safe according to today's standards? ;-) There are never too many times where we double-check our codebase for security issues. :pray:

**Pull request closed without merging**